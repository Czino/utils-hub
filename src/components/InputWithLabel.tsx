import { Input } from './Input'

type Props = React.ComponentProps<'input'> & {
  containerClassName?: string
  label: string
}
export const InputWithLabel = ({ containerClassName, label, id, ...props }: Props) => (
  <div className={containerClassName}>
    <label className="text-base" htmlFor={id}>
      {label}
    </label>
    <Input id={id} {...props} />
  </div>
)
