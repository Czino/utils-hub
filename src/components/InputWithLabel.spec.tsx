import { render } from '@testing-library/react'
import { InputWithLabel } from './InputWithLabel'

describe('InputWithLabel', () => {
  it('should render InputWithLabel with styles', () => {
    const { asFragment } = render(<InputWithLabel label="label" className="mt-4" value="text" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
