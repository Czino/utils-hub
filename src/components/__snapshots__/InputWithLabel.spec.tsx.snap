// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputWithLabel should render InputWithLabel with styles 1`] = `
<DocumentFragment>
  <div>
    <label
      class="text-base"
    >
      label
    </label>
    <input
      class="w-full text-xs md:text-xl appearance-none bg-white text-dark p-2 border border-dark rounded mt-4"
      value="text"
    />
  </div>
</DocumentFragment>
`;
